File: Ruby/Cluster.rb
class Cluster
  attr_reader :changing_words, :log_entries

  def initialize (signature, log_entries)
    changing_word_index = signature.varword_index
    changing_words = log_entries.map do |log_entry|
      log_entry.words[changing_word_index]
    end

    @changing_words = changing_words.uniq
    @log_entries = log_entries
  end
end

File: Node-js/Cluster.js
class Cluster {
  constructor(signature, log_entries) {
    let changing_word_index = signature.varword_index
    let changing_words = log_entries.map(log_entry => log_entry.words[changing_word_index])

    this.changing_words = [...new Set(changing_words)]
    this.log_entries = log_entries
  }
}

module.exports = Cluster
